sql/patch.sql

```diff
@@ -324,3 +324,7 @@ ALTER TABLE `users` ADD `portal_user` TINYINT(1) NOT NULL DEFAULT '0';
 #IfMissingColumn users supervisor_id
 ALTER TABLE `users` ADD `supervisor_id` INT(11) NOT NULL DEFAULT '0';
 #EndIf
+
+#IfMissingColumn documents document_data
+ALTER TABLE `documents` ADD `document_data` MEDIUMTEXT NULL DEFAULT NULL;
```

Comment:
no need to add the NULL DEFAULT NULL here. Just make it empty since MEDIUMTEXT defaults to NULL by default:
```
ALTER TABLE `documents` ADD `document_data` MEDIUMTEXT
```

Reply:
What a strange thing to nitpick. It's poor practice to not explicitly define default on column creation otherwise, it is implicite to the sql engine. While today it may be true, tomorrow, maybe not for Maria or Mysql! 
Which brings up now that Maria is becoming popular they have decided to depart from Mysql compatibility. This is huge for us if we keep trying to be compatible with every o.s and database engine out there. Example one: I wanted to use json data type however, Maria is incompatible with Mysqls json data type! We need to decide on one or the other so we can use the new capabilities afforded by either. Otherwise we're stuck in the past just so we can maintain support for both...

Reply:
This is just something I have been trained to see since trying to set default on TEXT fields has broken mariadb/mysql in past. This is why they have all been cleared to nothing in database.sql (one of the fun project i got to take on in the past)

Reply:
Regarding database, note we will be dropping mariadb 10.1 and mysql 5.6 support in new future, so can potentially use newer features (such as json, although by my reading of mariadb in past is that isn't truly use json object (mocked up mediumtext) whereas mysql is pure json object, which I am guessing where the incompatibilities on that lie)

Reply:
If we had to pick between mysql/mariadb, then mariadb would be the obvious choice. But hopefully we can keep supporting both (just gotta chew gum, walk, juggle, and swim at the same time; no biggie :) ).

Reply:
> trying to set default on TEXT fields has broken mariadb/mysql in past

That makes it a mysql/maria(my bet is on maria) bug and not syntax issue. Still, we'll go with your experience but i've never had issue with mysql in this regard.

Reply:
> If we had to pick between mysql/mariadb, then mariadb would be the obvious choice.

Good god, let's not have that debate!:) I've learned my lesson in that regard while working on an open source project i.e i'm weary!!

Reply:
here's my impression of a mysql8 developer:
"lets add a reserved keyword that breaks stuff in a minor release. ok, that was a bad idea, lets revert that in a couple more minor releases. I'm bored, lets break php with our default new shiny authentication..."
:)